flocker/node/script.py

```diff
@@ -179,11 +132,47 @@ def get_service(self, reactor, options):
         """
         return AgentLoopService(
             reactor=reactor,
-            deployer=self.deployer_factory(hostname=options["hostname"]),
+            deployer=self.deployer_factory(options=options),
             host=options["destination-host"], port=options["destination-port"],
         )
 
 
+def volume_service_from_config(config):
+    pass
+
+def dataset_deployer_from_options(options):
+    """
+    Returns an IDeployer configured by options.
+    """
+    import yaml
+    config = yaml.safe_load(options['config-file'].getContents())
```

Comment:
We probably want to separate out the parsing of the config-file, from building the services. Particularly since we'll want to do validation of the config-file.